### DISCUSSION TITLE
Is it possible to get the exact number of hits?

### DISCUSSION THREAD

**Discussion by dkalpakchi** [searchkit/searchkit#1264] [Q&A] [answered]
Hi everyone!

I want to get a total number of documents for each query (not the estimate of 10000). Normally I achieved this by using setting "track_total_hits" to true, but this doesn't seem to work for Searchkit? I've tried to redefine getQuery as follows:
```
getQuery: (query, search_attributes) => {
  return [
    "query": {
       "multi_match" : {
          "query":     query,
          "type":       "phrase",
          "fields":     search_attributes
        }
      },
      "track_total_hits": true
    }
  ];
},
```
This didn't work for me though, am I missing something or is it impossible currently?

Thank you in advance for your help!

**Comment by joemcelroy** [comment, marked as answer]
hi there!

Its possible but you need to use `beforeSearch` request hook instead to pass in `track_total_hits` to each request. 

The difference between `getQuery` and `beforeSearch` hooks is that `beforeSearch allows you to modify the whole request. `getQuery` is for those who want to adjust only the relevance of the organic query.  

https://www.searchkit.co/docs/api-documentation/searchkit#beforesearch-hook-function

Something like

```ts
client = SearchkitInstantsearchClient(sk, {
    hooks: {
        beforeSearch: async (searchRequests) => {

            return searchRequests.map((sr) => {
                return {
                  ...sr,
                  body: {
                    ...sr.body,
                    track_total_hits: true
                 }
              }
           })

        }
    }
})
```
